tokio/src/sync/watch.rs

```diff
@@ -112,6 +112,20 @@ pub struct Sender<T> {
 #[derive(Debug)]
 pub struct Ref<'a, T> {
     inner: RwLockReadGuard<'a, T>,
+    updated: bool,
+}
+
+impl<'a, T> Ref<'a, T> {
+    /// Indicates if the shared value has been detected as _changed_
+    /// **and** marked as seen.
+    ///
+    /// The result is `true` only if this reference has been obtained
+    /// from [`Receiver::borrow_and_update()`] in case changes have been
+    /// detected and marked as seen. In all other cases the result will
+    /// be `false`.
+    pub fn updated(&self) -> bool {
+        self.updated
+    }
```

Comment:
It seems more useful to have this function return whether the value was marked as seen before the call. Then the value is also useful for the `borrow` method.

Reply:
This would mean to
- add an invocation of `load()` to `borrow()` and
- rename the function to `has_changed()`.

I was unsure if the extra `load()` in `borrow()` is desired. It probably doesn't have any noticeable effect and a more consistent interface might be preferable.

Reply:
Done: https://github.com/tokio-rs/tokio/pull/4758/commits/d081f83f762c43cb826c70945a73ce3bb89f619f

I also added a comprehensive example.